internal/pubsub/pubsub_test.go

```diff
@@ -12,12 +12,17 @@ import (
 	"github.com/tendermint/tendermint/internal/pubsub"
 	"github.com/tendermint/tendermint/internal/pubsub/query"
 	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/types"
 )
 
 const (
 	clientID = "test-client"
 )
 
+type pubstring string
+
+func (pubstring) TypeTag() string { return "pubstring" }
```

Comment:
Can you explain to me what this is for or why you are doing this?

Reply:
This is the thing that makes the new (non-tmjson) handling work correctly, and I presume that it's here because the output of strings in the pubsub framework need to pass through the json encoding system as they exit the system.

Reply:
As @tychoish says. This PR requires event data to be more than just "anything" and in particular it has to support the type-tagged JSON encoding we use for variant interfaces. This type is just a shim around strings, so that the test cases (which previously just used them naked) can treat them as EventData.